services/transaction-statistics/shared/transactionStatistics.js

```diff
@@ -60,259 +55,170 @@ const getSelector = async (params) => {
-		.sort((a, b) => a.date.localeCompare(b.date)).reverse();
+				tokenStatsTimeline[tokenID] = timelineRaw.map((el) => ({
+					...el,
+					timestamp: Date.parse(el.date) / 1000,
+					transactionCount: parseInt(el.transactionCount, 10),
+				}));
+			}
+		},
+		{ concurrency: params.tokenIDs.length },
+	);
 
-	return orderedFinalResult;
+	return tokenStatsTimeline;
 };
 
 const getDistributionByAmount = async params => {
 	const db = await getDbInstance();
+	const tokenDistributionByAmount = {};
+
+	await BluebirdPromise.map(
+		params.tokenIDs,
+		async tokenID => {
+			const queryParams = await getSelector(params);
+			const result = (await db.find(
+				{
+					...queryParams,
+					whereIn: { property: 'tokenID', values: [tokenID, DB_CONSTANT.UNAVAILABLE] },
+				},
+				['amount_range', 'count'])).filter(o => o.count > 0);
+
+			const unorderedfinalResult = {};
+			result.forEach(entry => {
+				if (!unorderedfinalResult[entry.amount_range]) unorderedfinalResult[entry.amount_range] = 0;
+				unorderedfinalResult[entry.amount_range] += entry.count;
+			});
 
-	const result = (await db.find(await getSelector(params), ['amount_range', 'count'])).filter(o => o.count > 0);
-
-	const unorderedfinalResult = {};
-	result.forEach(entry => {
-		if (!unorderedfinalResult[entry.amount_range]) unorderedfinalResult[entry.amount_range] = 0;
-		unorderedfinalResult[entry.amount_range] += entry.count;
-	});
+			if (tokenID !== DB_CONSTANT.UNAVAILABLE) {
+				const orderedFinalResult = {};
+				Object.keys(unorderedfinalResult).sort().reverse()
+					.forEach(amountRange => {
+						orderedFinalResult[amountRange] = unorderedfinalResult[amountRange];
+					});
 
-	const orderedFinalResult = {};
-	Object.keys(unorderedfinalResult).sort().reverse()
-		.forEach(amountRange => orderedFinalResult[amountRange] = unorderedfinalResult[amountRange]);
+				tokenDistributionByAmount[tokenID] = orderedFinalResult;
+			}
+		},
+		{ concurrency: params.tokenIDs.length },
+	);
 
-	return orderedFinalResult;
+	return tokenDistributionByAmount;
 };
 
 const getDistributionByType = async params => {
 	const db = await getDbInstance();
 
-	const result = (await db.find(await getSelector(params), ['type', 'count'])).filter(o => o.count > 0);
+	const result = (await db.find(await getSelector(params), ['moduleCommand', 'count'])).filter(o => o.count > 0);
 
 	const unorderedfinalResult = {};
 	result.forEach(entry => {
-		if (!unorderedfinalResult[entry.type]) unorderedfinalResult[entry.type] = 0;
-		unorderedfinalResult[entry.type] += entry.count;
+		if (!unorderedfinalResult[entry.moduleCommand]) unorderedfinalResult[entry.moduleCommand] = 0;
+		unorderedfinalResult[entry.moduleCommand] += entry.count;
 	});
 
 	const orderedFinalResult = {};
-	Object.keys(unorderedfinalResult).sort((a, b) => Number(a) - Number(b))
-		.forEach(type => orderedFinalResult[type] = unorderedfinalResult[type]);
+	Object.keys(unorderedfinalResult)
+		.sort((a, b) => Number(a) - Number(b))
```

Comment:
Wouldn't the sorting fail because we now use strings instead of numbers? Please cross-check.